packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -87,10 +87,12 @@ Future<void> buildMacOS({
       'OBJROOT=${globals.fs.path.join(flutterBuildDir.absolute.path, 'Build', 'Intermediates.noindex')}',
       'SYMROOT=${globals.fs.path.join(flutterBuildDir.absolute.path, 'Build', 'Products')}',
       if (verboseLogging)
-        'VERBOSE_SCRIPT_LOGGING=YES',
+        'VERBOSE_SCRIPT_LOGGING=YES'
+      else
+        '-quiet',
       'COMPILER_INDEX_STORE_ENABLE=NO',
       ...environmentVariablesAsXcodeBuildSettings(globals.platform)
-    ], trace: true);
+    ], trace: false);
```

Comment:
Would love to combine more logic, iOS and macOS building should look almost the same.

Reply:
I'm confused here; IIRC you recently turned `trace` on for Windows and Linux to fix missing Dart failure output. Why do we want it for Windows and Linux but not macOS?

Reply:
trace only prints stderr by default, and forwards stdout to our verbose logging (hidden by default). with -quiet, xcode writes errors to stdout so we need to surface them.

> Why do we want it for Windows and Linux but not macOS?

Everything is doing something different with error logs 🕵️‍♂️

Reply:
Oh, I see. Could we use the new stdout error regex logic I added for MSBuild here, so the logic is more unified?

Reply:
with `-quiet` there shouldn't be any need, since only task stderr should be surfaced

Reply:
But there are other differences in the code flow. For instance, won't your current approach mean that errors are printed in the normal color on macOS, rather than red on every other platform? And any code, future or current, that hooks into the logger mechanism to know what the errors are won't see the macOS error output as errors.

Whereas if you pass `.*` as an error regex (along with using `-quiet`) then macOS errors will use the same path through the logger as errors on other platforms.

Reply:
Ahh, cool I didn't know it did that :)